src/git/gitProviderService.ts

```diff
@@ -655,6 +655,16 @@ export class GitProviderService implements Disposable {
 			if (added.length) {
 				// Defer the event trigger enough to let everything unwind
 				queueMicrotask(() => this.fireRepositoriesChanged(added));
+				for (const addedRepo of added) {
+					try {
+						await this.container.repositoryIdentity.addFoundRepositoryToMap(
+							addedRepo,
+							await this.container.repositoryIdentity.getRepositoryIdentity(addedRepo),
+						);
+					} catch (ex) {
+						Logger.error('Could not add discovered repository to identity map: ', ex);
+					}
+				}
```

Comment:
@eamodio What are your thoughts on me putting this logic in its own function and just using `queueMicrotask` for it here and in the other place below?

Reply:
Ended up doing it, and added gating: https://github.com/gitkraken/vscode-gitlens/pull/3633/commits/7130c687ca4c996a609dfa59c39d7772c392ad49